src/de.rs

```diff
@@ -397,6 +398,21 @@ where
         }
     }
 
+    fn recursion_checked_tagged<F, T>(&mut self, tag: u64, f: F) -> Result<T>
```

Comment:
This function is only called like this: `self.recursion_checked_tagged(tag.into(), |de| visitor.visit_newtype_struct(de))`. Therefore we don't need to pass in f but can just include it in the function.
Furthermore it should call `recursion_checked` and not reimplement it.

The new function should look like this: `fn handle_tagged_value<F, T, V>(&mut self, tag: u64, visitor: V) -> Result<V::Value>`